lib/inventory/inventory.go

```diff
@@ -162,6 +162,14 @@ func (h *downstreamHandle) tryRun(fn DownstreamCreateFunc, hello proto.UpstreamI
 }
 
 func (h *downstreamHandle) handleStream(stream client.DownstreamInventoryControlStream, upstreamHello proto.UpstreamInventoryHello) error {
+	agentMetadataCh := make(chan proto.UpstreamInventoryAgentMetadata, 1)
+	go func() {
+		agentMetadataCh <- fetchAgentMetadata(&fetchConfig{
```

Comment:
We can cache the result of this as part of `downstreamHandle`, no need to do this once per stream.

Something like
```go
type downstreamHandle struct {
	...

	agentMetadataDone chan struct {} // to be initialized with no buffer
	agentMetadata proto.UpstreamInventoryAgentMetadata
}

// in NewDownstreamHandle
go func() {
	h.agentMetadata = fetchAgentMetadata(...)
	close(h.agentMetadataDone)
}

// in handleStream
...
agentMetadataDone := h.agentMetadataDone
select  {
...
case <-agentMetadataDone:
	agentMetadataDone = nil
	if err := stream.Send(h.closeContext, h.agentMetadata); err != nil {
```

Reply:
Thanks!! Cool trick 😮 

Done in fd03367813142cb3f485288f507c2e1f7dee0545.